### DISCUSSION TITLE
How to train spacy transformer model with currently not avalilable languages [spacy-nightly]

### DISCUSSION THREAD

**Discussion by danielvasic** [explosion/spaCy#6319] [Help: Coding & Implementations] [answered]
Hello, I would like to train transformer model for Croatian language. I have converted the pyTorch BERT model to spaCy model but when I try to add transformer pipeline in the cfg file the error "Transformer model is not available for language hr". I was hoping that the installation of the BERT model and linking it to spaCy data directory with hr shortcut will solve the problem but that didn't help. Any suggestions? 

I have successfully trained the tok2vec model with init_tok2vec using BERT embeddings but the accuracy of the model remains the same obvously I have missinterpreted the usage for this component. 

## Your Environment

* Operating System: Ubuntu 18.04 
* Python Version Used: 3.7.2
* spaCy Version Used: spacy-nightly 3.0.0rc1

**Comment by adrianeboyd** [comment]
The transformer model needs to be something that you can load with `AutoModel.from_pretrained()`. This can be the name of a model provided by huggingface like "roberta-base" (so a model you can find in https://huggingface.co/models) or it can be the path to a local model directory.

You no longer need a step where you're "converting the model to a spacy model" as in 0.x versions of `spacy-transformers`.

**Comment by danielvasic** [comment]
Dear @adrianeboyd ,

Thank you for the response, how can I define the configuration file for using pretrained [CroSloEngualBert](https://huggingface.co/EMBEDDIA/crosloengual-bert) or is this something that I cannot yet define by using CLI, if not how can I train this model for parsing, tagging and ner.

**Comment by adrianeboyd** [comment, marked as answer]
I haven't tested this, but the config would look like:

```ini
[components.transformer.model]
@architectures = "spacy-transformers.TransformerModel.v1"
name = "EMBEDDIA/crosloengual-bert"
```

You can run `spacy init config` with the English transformer defaults (it will fill in `en` and `roberta-base`) and then adjust `nlp.lang` and the transformer model in the config.

**Comment by danielvasic** [comment]
Dear @adrianeboyd ,

Thanks very much for the instructions will try and report back if there are any issues. 

Best regards,
Daniel
